### PR TITLE
Update .Net framework versions

### PR DESCRIPTION
**Pull request by dsmithson**
- Kleinware.LikeType updated to target .Net Standard 2.0
- Example and test projects updated to .Net 8.0

### CODE REVIEW

**Comment by dsmithson** [comment]
Hello there!  I know this repo hasn't been touched in quite a while, but I use this and recently updated my solution to target .Net core.  As part of this I forked your repo and updated the Kleinware/LikeType project to target .Net Standard 2.0 so it can be used with both the full .Net Framework and .Net Core.  

I figured I'd pay it forward and see if you were interested in incorporating this change into your repo and maybe even update the Nuget package out there.  No pressure of course.  Thanks a ton for making this repo!

File: Kleinware.LikeType/Kleinware.LikeType.csproj [hunk 1]
```diff
@@ -1,47 +1,17 @@
...
-    <FileAlignment>512</FileAlignment>
-    <TargetFrameworkProfile />
+    <PackageId>Kleinware.LikeType</PackageId>
+    <Version>4.0.0</Version>
+    <Authors>Matt Klein</Authors>
+    <Company>Matt Klein</Company>
+    <Description>Provides 'typedef' like behavior to simple C# classes.</Description>
+    <Copyright>Copyright 2024</Copyright>
+    <PackageTags>typedef implicit operator immutable</PackageTags>
+    <PackageProjectUrl>https://github.com/dsmithson/LikeType</PackageProjectUrl>
```
Comment (kleinwareio) [inline Kleinware.LikeType/Kleinware.LikeType.csproj]: Could you change this back to `https://github.com/mattklein999/LikeType`?

**Review by kleinwareio — COMMENTED** [review]
Thanks Derek! If you could update the URL then I'll merge this in and publish the new version on NuGet.

**Review by kleinwareio — COMMENTED** [review]
Sorry, could you also update the version in the `Requirements` section of the `README.me`?

File: Kleinware.LikeType/Kleinware.LikeType.csproj (hunk 1 above)
Comment (dsmithson) [reply Kleinware.LikeType/Kleinware.LikeType.csproj]: Ooh dang great catch!  Updated!

**Comment by dsmithson** [comment]
Another great catch on the README - take a peek at the update I put in there.  Thanks again!

**Comment by kleinwareio** [comment]
Thanks again Derek! I've also updated NuGet https://www.nuget.org/packages/Kleinware.LikeType/4.0.0

**Pull request merged**